test/unit/player-loadmedia.test.js

```diff
@@ -0,0 +1,152 @@
+/* eslint-env qunit */
+import sinon from 'sinon';
+import TestHelpers from './test-helpers';
+
+QUnit.module('Player: loadMedia/getMedia', {
+
+  beforeEach() {
+    this.clock = sinon.useFakeTimers();
+    this.player = TestHelpers.makePlayer({});
+  },
+
+  afterEach() {
+    this.player.dispose();
+    this.clock.restore();
+  }
+});
+
+QUnit.test('loadMedia sets source from a string', function(assert) {
+  this.player.loadMedia({
+    src: 'foo.mp4'
+  });
+
+  assert.strictEqual(this.player.currentSrc(), 'foo.mp4', 'currentSrc was correct');
+});
+
+QUnit.test('loadMedia sets source from an object', function(assert) {
+  this.player.loadMedia({
+    src: {
+      src: 'foo.mp4',
+      type: 'video/mp4'
+    }
+  });
+
+  assert.strictEqual(this.player.currentSrc(), 'foo.mp4', 'currentSrc was correct');
+});
+
+QUnit.test('loadMedia sets source from an array', function(assert) {
```

Comment:
might be worth adding a test that has several sources and then `currentSources` should return those multiple sources.